src/test.jl

```diff
@@ -41,23 +41,54 @@ function check_coherent_types(O::Type, tow::TestObjectWrapper)
 end
 
 """
-    test(::Type{<:Interface}, obj)
+    test(m::Module)
+    test(::Type{<:Interface}, m::Module)
+    test(::Type{<:Interface}, obj::Type)
 
 Test if an interface is implemented correctly for an object,
 returning `true` or `false`.
 
 If no interface type is passed, Interfaces.jl will find all the
 interfaces available and test them.
 """
-function test(T::Type{<:Interface{Keys}}, O::Type, test_objects; kw...) where Keys
+function test(T::Type{<:Interface}, mod::Module; kw...) 
+    methodlist = methods(implements, Tuple{T,<:Any})
+    _test_module(mod, methodlist; kw...)
+end
+function test(mod::Module; kw...) 
+    methodlist = methods(implements, Tuple{<:Any,<:Any})
+    _test_module(mod, methodlist; kw...)
+end
+
+
+function _test_module(mod, methodlist; kw...)
```

Comment:
I have zero clue what this function does. Send help

Reply:
Ok, I'll add a comment! But yes this is where we dig into some internals and use the method table as the global state of all available implementations.

Its looping over the methods in the methodlist from `methods(implementes `Tuple{T, <:Any})` and checking which of them the `implements` method is in `mod`, then running `test` on the type signature, that we have to reconstruct a little.